pkg/nfd-master/nfd-master.go

```diff
@@ -277,13 +287,67 @@ func (m *nfdMaster) Run() error {
 			if m.nfdController != nil && m.args.EnableNodeFeatureApi {
 				m.nfdController.updateAllNodesChan <- struct{}{}
 			}
+			// Restart the node updater pool
+			m.stopNodeUpdaterPool()
+			go m.runNodeUpdaterPool()
+
 		case <-m.stop:
 			klog.Infof("shutting down nfd-master")
 			return nil
 		}
 	}
 }
 
+func (m *nfdMaster) processNodeLabelRequest() bool {
+	nodeName, quit := m.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer m.queue.Done(nodeName)
+
+	if err := m.nfdAPIUpdateOneNode(nodeName.(string)); err != nil {
+		if m.queue.NumRequeues(nodeName) < 5 {
+			klog.Infof("retrying labeling request for node %s", nodeName)
+			m.queue.AddRateLimited(nodeName)
+			return true
+		} else {
+			klog.Errorf("error labeling node %v: %v", nodeName, err)
+		}
+	}
+	m.queue.Forget(nodeName)
+	return true
+}
+
+func (m *nfdMaster) runNodeUpdater() {
+	for {
+		select {
+		case <-m.nodeUpdaterPoolStop:
+			return
+		default:
+			m.processNodeLabelRequest()
+		}
+	}
+}
+
+func (m *nfdMaster) runNodeUpdaterPool() {
+	klog.Infof("Starting the NFD master node updater pool with %v parallel updaters", m.config.NfdApiParallelism)
+	m.nodeUpdaterPoolStop = make(chan struct{}, 1)
+	m.queue = workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	defer m.queue.ShutDown()
+	for i := 0; i < m.config.NfdApiParallelism; i++ {
+		go m.runNodeUpdater()
+	}
+	<-m.nodeUpdaterPoolStop
+	klog.Info("Stopping the NFD master node updater pool")
+}
+
+func (m *nfdMaster) stopNodeUpdaterPool() {
+	// notifies all node updaters goroutines to stop and shut downs the node update pool
+	close(m.nodeUpdaterPoolStop)
```

Comment:
Hmm, I think ideally we should wait for the updater goroutines to finish (waitgroup). To be on the safe side, even if an unlikely race. We might e.g. have in-flight retries that get intermixed with node updates from the new pool after re-configuration.

One idea: `runNodeUpdaterPool()` could be changed to non-blocking `startNodeUpdaterPool` that just creates the worker goroutines and adds them to a waitgroup and returns. Then `stopNodeUpdaterPool` would signal these worker goroutines to stop and wait for them to finish (with the waitgroup). 

Also, I don't think we actually need the stop channel for anything. 

```suggestion
	m.queue.ShutDown()
```